### PR TITLE
Follow up improvements for `bcsymbolmap` paths

### PR DESCRIPTION
**Pull request by dnkoutso**
Follow up on https://github.com/CocoaPods/CocoaPods/pull/8470 /cc @dacaiguoguogmail

### CODE REVIEW

File: spec/unit/target/pod_target_spec.rb [hunk 1]
```diff
@@ -662,6 +662,39 @@ module Pod
         end
       end
 
+      describe 'resource and framework paths' do
```
Comment (dnkoutso) [inline spec/unit/target/pod_target_spec.rb]: moved these to its own describe method, they should not have been under "test spec support"

File: lib/cocoapods/target/framework_paths.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@ def hash
       end
 
       def all_paths
-        [source_path, dsym_path].compact
+        [source_path, dsym_path, bcsymbolmap_paths].compact
```
Comment (dnkoutso) [inline lib/cocoapods/target/framework_paths.rb]: @amorde do you think we want this? It makes perfect sense for this class but this propagates to input paths of script phases...As I am typing this I think we need it and that was a bug in the original PR.

File: spec/unit/target/pod_target_spec.rb [hunk 3]
```diff
@@ -662,6 +662,50 @@ module Pod
...
+            ],
+            'WatermelonLib/Tests' => [],
+            'WatermelonLib/SnapshotTests' => [],
+            'WatermelonLib/App' => [
+              Target::FrameworkPaths.new('${BUILT_PRODUCTS_DIR}/WatermelonLib/WatermelonLib.framework'),
+            ],
+          }
+        end
+
+        it 'returns correct vendored framework paths' do
```
Comment (dnkoutso) [inline spec/unit/target/pod_target_spec.rb]: new test for vendored framework paths.

File: lib/cocoapods/target/framework_paths.rb (hunk 2 above)
Comment (amorde) [reply lib/cocoapods/target/framework_paths.rb]: yes definitely, I had thought this was covered in the review but I guess it was just included in `hash` https://github.com/CocoaPods/CocoaPods/pull/8470#discussion_r252100005

Comment (amorde) [reply lib/cocoapods/target/framework_paths.rb]: could you add a test for this? just realized there aren't any

Comment (dnkoutso) [reply lib/cocoapods/target/framework_paths.rb]: yes sir!

Comment (dnkoutso) [reply lib/cocoapods/target/framework_paths.rb]: added

**Pull request merged**